### ISSUE TITLE
LLVM does not support SELECT without FROM currently

### ISSUE THREAD

**Issue by hzxa21** [cmu-db/peloton#907]
The LLVM engine does not support SELECT without FROM right now because it requires the projection plan to have a child plan. Here is the check:
https://github.com/cmu-db/peloton/blob/c30cb3137b4aca4539d7681cec824f60a0789c55/src/codegen/operator/projection_translator.cpp#L26

If we use SELECT 1 FROM A where A is an empty table, the result will be empty. The reason might be that if the table is empty, the LLVM will not even generate codes for the projection logics.

**Comment by tcm-marcel** [comment]
Postgres also returns an empty result: http://sqlfiddle.com/#!17/5bfca/1/0

The codegen will generate code that evaluates the expression `1` once per row. 
Empty table -> no rows to run the expression on -> empty result

**Comment by pmenon** [comment]
I agree with @tcm-marcel. The plan in that form (i.e., `select 1 from A`) has a scan over table A, and thus, the projection plan must have a child. Note that `select 1 from A` has different semantics that `select 1`: The former returns `1` *for every row* in `A`, while the latter returns only a single row `1`.

**Comment by prashasthip** [comment]
[https://github.com/cmu-db/peloton/blob/c30cb3137b4aca4539d7681cec824f60a0789c55/src/codegen/query_compiler.cpp#L58](url)

This prevents executing queries like:

select udf(parameter);

**Comment by hzxa21** [comment]
@prashasthip You are right. I think the reason why we do this check is that codegen currently cannot execute ProjectionPlan with no children.
